src/object/srv_obj_migrate.c

```diff
@@ -1152,7 +1152,7 @@ __migrate_fetch_update_parity(struct migrate_one *mrone, daos_handle_t oh,
 			ptr += size * iods[i].iod_size;
 			offset = recx->rx_idx;
 			size = recx->rx_nr;
-			parity_eph = ephs[i][j];
+			parity_eph = encode ? ephs[i][j] : mrone->mo_epoch;
```

Comment:
@jolivier23 FYI, this is the key change which fixed the data corruption issue on our side.

Reply:
Yeah, I backported this whole patch.  Had to pull in a couple of others but with this, I'm not seeing corruption either

https://github.com/daos-stack/daos/pull/14519